### DISCUSSION TITLE
Async Execution Cancellation

### DISCUSSION THREAD

**Discussion by adiamante** [yantrajs/yantra#69] [Ideas]
Hi would you be open to adding a cancellation token for async execution?

I'm looking for something like the following:
```
var context = new JSContext();
var cancellationTokenSource = new CancellationTokenSource();
await context.ExecuteAsync("code", cancellationTokenSource .Token);

// to stop execution/throw cancelled exception
cancellationTokenSource.Cancel();
```

**Comment by ackava** [comment]
@adiamante Currently not possible, but we are adding `AbortController` and you will be able to pass CancellationToken as `AbortController` which can be used internally to pause some of async operations. However, the issue is currently in CLR, we have no way of pausing execution of IL instructions so code will not stop execution. But any IO utilizing cancellation token will throw an exception which will then throw AbortError inside JavaScript.

**Reply by adiamante** [reply]
@ackava Understood. Thank you for the reply
